### PR TITLE
Updated ENLSVG variables to use snake_case

### LINKED ISSUE #2716: Update variables in ENLSVG to use Snake Case
### Description of the task

ENLSVG is an external path planning implementation we use and we'd like to maintain our own copy of it. Currently, the external implementation doesn't follow our code guidelines and we'd like to update it so that it does.

One major problem is that the external implementation uses camelCase for variable names while we use snake_case. This ticket would involve converting these variables into snake case. For example (`ticketCheck` would become `ticket_check`and `parentPtrs` would become `parent_ptrs`)

Can refer to #1939 for more detail and a discussion.

### Acceptance criteria

- [ ] Navigate to Software/src/extlibs/enlsvg/Pathfinding/ and Software/src/extlibs/enlsvg/src to look at the respective header and cpp files
- [ ] Identify offending variables and update them to use snake case
- [ ] Fix any build issues that come up when running `bazel build //software/...` in the Software/src directory

### PR DESCRIPTION
**Pull request by sauravbanna**
### Description
All camelCase variables in Software/src/extlibs/enlsvg/Pathfinding/ and Software/src/extlibs/enlsvg/src changed to use snake_case

### Testing Done
Builds successfully, all tests pass

### Resolved Issues

Resolves #2716

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue

### CODE REVIEW

File: src/extlibs/enlsvg/Pathfinding/ENLSVGGraph.h [hunk 1]
```diff
@@ -17,13 +17,13 @@ namespace Pathfinding
...
-            const VertexID destVertex;
+            const VertexID source_vertex;
+            const VertexID dest_vertex;
             int level;
-            std::vector<EdgeID> tautOutgoingEdges;
+            std::vector<EdgeID> taut_outgoing_edges;
 
             EdgeData(VertexID sourceVertex, VertexID destVertex, int level)
-                : sourceVertex(sourceVertex), destVertex(destVertex), level(level)
+                : source_vertex(sourceVertex), dest_vertex(destVertex), level(level)
```
Comment (nimazareian) [inline src/extlibs/enlsvg/Pathfinding/ENLSVGGraph.h]: `sourceVertex` and `destVertex` should be updated, similar with multiple other function parameters in this file (e.g. `imNext` and `imLast`)

File: src/extlibs/enlsvg/src/IndirectHeap.cpp [hunk 2]
```diff
@@ -5,84 +5,84 @@
...
+        if (heap_size == 0)
             std::cout << "ERROR: EMPTY HEAP!" << std::endl;
-        else if (heapSize == 1)
+        else if (heap_size == 1)
         {
-            --heapSize;
+            --heap_size;
             return getOut(0);
         }
         // nodeList.size() > 1
```
Comment (nimazareian) [inline src/extlibs/enlsvg/src/IndirectHeap.cpp]: I know that this is not part of this ticket, but could you also remove the commented out code here?

**Review by nimazareian — CHANGES_REQUESTED** [review]
Overall looks good to me! Just noticed a couple of variables that should still be updated.

File: src/software/thunderscope/robot_communication.py [hunk 3]
```diff
@@ -58,18 +58,18 @@ def __init__(
...
+        #     raise Exception("Could not find estop, make sure its plugged in")
 
     def __send_estop_state(self):
         while True:
-            self.full_system_proto_unix_io.send_proto(
-                EstopState, EstopState(is_playing=self.estop_reader.isEstopPlay())
-            )
+            # self.full_system_proto_unix_io.send_proto(
+            #     EstopState, EstopState(is_playing=self.estop_reader.isEstopPlay())
+            # )
```
Comment (nimazareian) [inline src/software/thunderscope/robot_communication.py]: It looks like some of the code was accidentally commented out!?

File: src/extlibs/enlsvg/Pathfinding/ENLSVGGraph.h (hunk 1 above)
Comment (sauravbanna) [reply src/extlibs/enlsvg/Pathfinding/ENLSVGGraph.h]: done

File: src/extlibs/enlsvg/src/IndirectHeap.cpp (hunk 2 above)
Comment (sauravbanna) [reply src/extlibs/enlsvg/src/IndirectHeap.cpp]: done

File: src/extlibs/enlsvg/Pathfinding/LineOfSightScanner.h [hunk 4]
```diff
@@ -76,37 +76,37 @@ namespace Pathfinding
                                 : left_down_extents[(y + 1) * extents_size_x + x_L];
         }
 
-        inline int leftDownExtent(int xL, int y) const
+        inline int leftDownExtent(int x_L, int y) const
```
Comment (nimazareian) [inline src/extlibs/enlsvg/Pathfinding/LineOfSightScanner.h]: I think the `L` in these variables should be made small. But to also make it a bit more readable, could you replace the instances of `x_L` to be `x_left` and similarly for `x_R`?

**Review by nimazareian — APPROVED** [review]

**Review by nimazareian — CHANGES_REQUESTED** [review]
oops, I meant to request changes

File: src/software/thunderscope/robot_communication.py (hunk 3 above)
Comment (sauravbanna) [reply src/software/thunderscope/robot_communication.py]: Sorry about that, I was working on another issue and accidentally committed that file.

File: src/extlibs/enlsvg/Pathfinding/LineOfSightScanner.h (hunk 4 above)
Comment (sauravbanna) [reply src/extlibs/enlsvg/Pathfinding/LineOfSightScanner.h]: done

File: src/extlibs/enlsvg/Pathfinding/LineOfSightScanner.h [hunk 5]
```diff
@@ -70,61 +70,65 @@ namespace Pathfinding
...
+        void generateUpwards(ScannerStacks& data, Rational left_bound,
+                             Rational right_bound, int sx, int sy, int curr_y,
+                             bool left_inclusive, bool right_inclusive) const;
+        void generateDownwards(ScannerStacks& data, Rational left_bound,
+                               Rational right_bound, int sx, int sy, int curr_y,
+                               bool left_inclusive, bool right_inclusive) const;
+        void generateAndSplitIntervals(ScannerStacks& data, int check_y, int new_y,
+                                       int sx, int sy, Rational left_bound,
+                                       Rational right_bound, bool left_inclusive,
+                                       bool right_inclusive) const;
```
Comment (williamckha) [inline src/extlibs/enlsvg/Pathfinding/LineOfSightScanner.h]: Should be s_x and s_y

File: src/extlibs/enlsvg/Pathfinding/LineOfSightScanner.h [hunk 6]
```diff
@@ -70,61 +70,65 @@ namespace Pathfinding
...
-        std::vector<int> rightDownExtents;
-        std::vector<int> leftDownExtents;
+        std::vector<int> right_down_extents;
+        std::vector<int> left_down_extents;
 
         void generateTautDirectionStartingStates(ScannerStacks& data, int sx,
                                                  int sy) const;
         void generateAllDirectionStartingStates(ScannerStacks& data, int sx,
                                                 int sy) const;
         void exploreStates(ScannerStacks& data, int sx, int sy) const;
```
Comment (williamckha) [inline src/extlibs/enlsvg/Pathfinding/LineOfSightScanner.h]: Should be s_x and s_y

File: src/extlibs/enlsvg/Pathfinding/LineOfSightScanner.h [hunk 7]
```diff
@@ -70,61 +70,65 @@ namespace Pathfinding
         void computeTautDirNeighbours(ScannerStacks& data, int sx, int sy) const;
```
Comment (williamckha) [inline src/extlibs/enlsvg/Pathfinding/LineOfSightScanner.h]: Should be s_x and s_y

**Review by williamckha — CHANGES_REQUESTED** [review]
LGTM! Just a few things

**Review by nimazareian — COMMENTED** [review]
Other than William's comments, it looks good!

File: src/extlibs/enlsvg/Pathfinding/LineOfSightScanner.h (hunk 6 above)
Comment (sauravbanna) [reply src/extlibs/enlsvg/Pathfinding/LineOfSightScanner.h]: done

File: src/extlibs/enlsvg/Pathfinding/LineOfSightScanner.h (hunk 7 above)
Comment (sauravbanna) [reply src/extlibs/enlsvg/Pathfinding/LineOfSightScanner.h]: done

**Review by nimazareian — APPROVED** [review]
LGTM! 💯

**Review by williamckha — APPROVED** [review]

**Pull request merged**